monkestation/code/modules/antagonists/bloodling/abilities/devour.dm

```diff
@@ -0,0 +1,50 @@
+/datum/action/cooldown/mob_cooldown/bloodling/devour
+	name = "Devour Limb"
+	desc = "Allows you to consume a creatures limb."
+	button_icon_state = "alien_hide"
+	cooldown_time = 20 SECONDS
+
+/datum/action/cooldown/mob_cooldown/bloodling/devour/PreActivate(atom/target)
+	var/mob/living/mob = target
+	if(!iscarbon(mob))
+		owner.balloon_alert(owner, "only works on carbons!")
+		return FALSE
+	return ..()
+
+/datum/action/cooldown/mob_cooldown/bloodling/devour/Activate(atom/target)
+	StartCooldown()
+	return eat_limb(target)
+
+/datum/action/cooldown/mob_cooldown/bloodling/devour/proc/eat_limb(atom/target)
+	var/mob/living/basic/bloodling/our_mob = owner
+	var/list/candidate_for_removal = list()
+	var/mob/living/carbon/carbon_target = target
+
+	// Loops over the limbs of our target carbon, this is so stuff like the head, chest or unremovable body parts arent destroyed
+	for(var/obj/item/bodypart/bodypart in carbon_target.bodyparts)
+		if(bodypart.body_zone == BODY_ZONE_HEAD)
+			continue
+		if(bodypart.body_zone == BODY_ZONE_CHEST)
+			continue
+		if(bodypart.bodypart_flags & BODYPART_UNREMOVABLE)
+			continue
+		candidate_for_removal += bodypart.body_zone
+
+	if(!length(candidate_for_removal))
+		return FALSE
+
+	var/limb_to_remove = pick(candidate_for_removal)
+	var/obj/item/bodypart/target_part = carbon_target.get_bodypart(limb_to_remove)
+
+	if(isnull(target_part))
+		return FALSE
+
+	target_part.dismember()
+	qdel(target_part)
+	our_mob.add_biomass(20)
```

Comment:
this seems like a really large amount of biomass for this, even more so as this is really strong in combat due to it being an instant free dismember

Reply:
I'll halve it